craft_application/util/paths.py

```diff
@@ -39,3 +41,17 @@ def get_managed_logpath(app: AppMetadata) -> pathlib.PosixPath:
 def get_filename_from_url_path(url: str) -> str:
     """Get just the filename of a URL path."""
     return pathlib.PurePosixPath(urllib.parse.urlparse(url).path).name
+
+
+def get_work_dir(project_dir: pathlib.Path) -> pathlib.Path:
+    """Get the work directory to hand to craft-parts.
+
+    :param project_dir: The directory containing the project
+    :returns: The craft-parts work directory.
+
+    When running in managed mode, this returns the managed mode work directory
+    of ``/root``. Otherwise, it returns the project directory.
+    """
+    if is_managed_mode():
+        return pathlib.Path("/root")
```

Comment:
It would be nice to make this a constant in the future, we refer to it in the application and provider service too.

Reply:
Yeah my goal is to remove those altogether and replace them with calls here.